Sources/Clappr_iOS/Classes/Animation/DoubleTapAnimation.swift

```diff
@@ -0,0 +1,254 @@
+import UIKit
+
+class DoubleTapAnimation {
+    private var backLabel = UILabel()
+    private var fowardLabel = UILabel()
+    private var core: Core?
+    
+    private let playImage = UIImage.fromName("play", for: PlayButton.self)
+    
+    private lazy var backIcon1 = mirrorImage(UIImageView(image: playImage))
+    private lazy var backIcon2 = mirrorImage(UIImageView(image: playImage))
+    private lazy var backIcon3 = mirrorImage(UIImageView(image: playImage))
+    
+    private lazy var fowardIcon1 = UIImageView(image: playImage)
+    private lazy var fowardIcon2 = UIImageView(image: playImage)
+    private lazy var fowardIcon3 = UIImageView(image: playImage)
+    
+    private var leftBubbleView = UIView()
+    private var rightBubbleView = UIView()
+    
+    private var leftBubbleHeight = NSLayoutConstraint()
+    private var leftBubbleWidth = NSLayoutConstraint()
+    
+    private var rightBubbleHeight = NSLayoutConstraint()
+    private var rightBubbleWidth = NSLayoutConstraint()
+    
+    init(_ core: Core?) {
+        self.core = core
+        setup(core)
+    }
+    
+    func animateBackward() {
+        guard let playback = core?.activePlayback,
+            playback.position - 10 > 0.0,
+            let view = core?.view else { return }
+        animate(leftBubbleView, parentView: view, widthConstraint: leftBubbleWidth, heightConstraint: leftBubbleHeight)
+        animate(backLabel)
+        animate(backIcon3, delay: 0)
+        animate(backIcon2, delay: 0.2)
+        animate(backIcon1, delay: 0.4)
+    }
+    
+    func animateForward() {
+        guard let playback = core?.activePlayback,
+            playback.position + 10 < playback.duration,
```

Comment:
What do you think about moving this check to the `Plugin` class? 
OI have the impression it has more in common with plugin rules than it has to do with `Animation`.

Reply:
Totally! good catch!